monasca_rsyslog/main.py

```diff
@@ -23,30 +23,31 @@
 sys.stdout = open(STDOUT_FILE, 'a')
 sys.stderr = open(STDERR_FILE, 'a')
 
-print('Started')
-sys.stdout.flush()
-
 from client import Client
+from select import select
+
+def stdin_by_line(poll_interval):
+    """ Helper for performing line-by-line reads of stdin. """
 
-def stdin_by_line():
-    """Helper for performing line-by-line reads of stdin."""
-    
     while True:
-        line = sys.stdin.readline()
-        if line:
-            yield line
+        buffer_is_not_empty, _, _ = select([sys.stdin], [], [], poll_interval)
```

Comment:
Interesting approach!

Reply:
Thanks, seems to work surprisingly well!